relay-server/src/services/processor.rs

```diff
@@ -1391,37 +1404,58 @@ impl EnvelopeProcessorService {
         &self,
         state: &mut ProcessEnvelopeState<G>,
     ) -> Result<(), ProcessingError> {
-        let options = &self.inner.global_config.current().options;
+        let attachment_type = state
+            .envelope()
+            .get_item_by(|item| item.attachment_type().is_some())
+            .and_then(|item| item.attachment_type())
+            .map(|ty| ty.to_string())
+            .unwrap_or("none".to_string());
```

Comment:
Let's not allocate in the default case.
```suggestion
            ;
```